contrib/inventory/ec2.py

```diff
@@ -789,9 +793,19 @@ def add_instance(self, instance, region):
             else:
                 hostname = getattr(instance, self.hostname_variable)
 
+        # set the hostname from route53
+        if self.route53_enabled and self.route53_hostnames:
+            route53_names = self.get_instance_route53_names(instance)
+            for name in route53_names:
+                if name.endswith(self.route53_hostnames):
+                    hostname = name
+
         # If we can't get a nice hostname, use the destination address
         if not hostname:
             hostname = dest
+        # to_safe strips hostname characters like dots, so don't strip if using route53 hostnames
+        elif self.route53_enabled and self.route53_hostnames:
+            hostname = hostname.lower()
```

Comment:
This conditional needs to only cover hostnames that actually match the route53 name. If I have some instances with example.com suffixes, and some without, then those other hostnames won't be passed through to_safe. So this check also needs something like `and hostname.endswith(self.route53_hostnames)` to only operate on hostnames that actually match DNS.